plugin-server/src/cdp/hog-transformations/hog-transformer.service.ts

```diff
@@ -288,15 +276,20 @@ export class HogTransformerService {
                     transformationsSucceeded.push(transformationIdentifier)
                 }
 
-                // Observe the results to update rate limiting state - only if feature flag is enabled
-                if (results.length > 0 && shouldRunHogWatcher) {
-                    try {
-                        const timer = transformEventHogWatcherLatency.startTimer({ operation: 'observeResults' })
-                        await this.hogWatcher.observeResults(results)
-                        timer()
-                    } catch (error) {
-                        logger.warn('⚠️', 'HogWatcher observeResults failed', { error })
-                    }
+                // Observe the results to update degraded state of HogFunctions
+                const watcherPromises: Promise<void>[] = []
+                if (shouldRunHogWatcher) {
+                    const timer = hogWatcherLatency.startTimer({ operation: 'observeResults' })
+                    const watcherPromise = this.hogWatcher
+                        .observeResults(results)
+                        .then(() => {
+                            timer() // Stop the timer when the promise resolves
+                        })
+                        .catch((error) => {
+                            timer() // Make sure to stop the timer even if there's an error
+                            logger.warn('⚠️', 'HogWatcher observeResults failed', { error })
+                        })
+                    watcherPromises.push(watcherPromise)
```

Comment:
I think this muddies the function a little... you don't need to observe here (in fact I would argue we don't want to) as this is the "pure" part of the function that doesn't actually write data to other sources like redis or capture.

I think instead, it should just return the result and the calling of the observeResults should be done in transformEventAndProduceMessages, that way it can just be given back as part of messagePromises instead of adding the watcherPromises (there is no difference between them from the callers perspective. They are just "delayed work to be awaited for later"